components/analytics/compact-dashboard.tsx

```
@@ -0,0 +1,1333 @@
+
+                                                    // Convert to array and sort based on user preference
+                                                    const sortedGroups = Object.values(errorGroups).sort((a, b) => {
+                                                        switch (sortPreference) {
+                                                            case 'count':
+                                                                // Sort by number of occurrences (descending)
+                                                                return b.bots.length - a.bots.length;
+
+                                                            case 'priority':
+                                                                // Sort by priority (critical > high > medium > low)
+                                                                const priorityOrder = { 'critical': 0, 'high': 1, 'medium': 2, 'low': 3, undefined: 4 };
+                                                                const priorityA = priorityOrder[a.priority as keyof typeof priorityOrder] || 4;
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)